packages/vm/lib/evm/opFns.ts

```diff
@@ -806,9 +773,6 @@ export const handlers: { [k: string]: OpHandler } = {
     runState.eei.finish(returnData)
   },
   REVERT: function (runState: RunState) {
-    if (!runState._common.gteHardfork('byzantium')) {
-      trap(ERROR.INVALID_OPCODE)
-    }
     const [offset, length] = runState.stack.popN(2)
     subMemUsage(runState, offset, length)
     let returnData = Buffer.alloc(0)
```

Comment:
Had a short look here to give this an additional thought if there might be side effects (conclusion: no).